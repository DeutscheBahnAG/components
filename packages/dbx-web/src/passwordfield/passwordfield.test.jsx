import React from 'react';
import { mount } from 'enzyme';
import Passwordfield from './passwordfield-doc';

const noopFn = () => {};

describe('Passwordfield component', () => {
  it('should render a text input', () => {
    const wrapper = mount(<Passwordfield value="test123" onChange={noopFn} />);
    expect(wrapper.find('input[type="password"]')).toHaveLength(1);
  });

  it('should disable the text input and the button', () => {
    const wrapper = mount(<Passwordfield disabled value="test123" />);
    const field = wrapper.find('input[type="password"]');
    expect(field).toHaveLength(1);
    expect(field.prop('disabled')).toBe(true);
    expect(wrapper.find('.dbx-button--disabled')).toHaveLength(1);
  });

  it('should disable the button when the value is empty', () => {
    const wrapper = mount(<Passwordfield value="" />);
    expect(wrapper.find('.dbx-button--disabled')).toHaveLength(1);
  });

  it('should enable the button when the value is set', () => {
    const wrapper = mount(<Passwordfield value="test123" />);
    expect(wrapper.find('.dbx-button:not(.dbx-button--disabled)')).toHaveLength(1);
  });
});
