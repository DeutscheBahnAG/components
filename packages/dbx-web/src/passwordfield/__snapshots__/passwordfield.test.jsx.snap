// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Passwordfield component should disable the button when the value is empty 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--password dbx-textfield--xl"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value=""
    />
    <span
      class="dbx-textfield-suffix"
    >
      <button
        class="dbx-button dbx-button--hover-only dbx-button--round dbx-button--disabled dbx-button--s"
        disabled=""
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="dbx-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="dbx-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should disable the text input and the button 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--password dbx-textfield--xl dbx-textfield--filled dbx-textfield--disabled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      disabled=""
      spellcheck="false"
      type="password"
      value="test456"
    />
    <span
      class="dbx-textfield-suffix"
    >
      <button
        class="dbx-button dbx-button--hover-only dbx-button--round dbx-button--disabled dbx-button--s"
        disabled=""
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="dbx-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="dbx-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should enable the button when the value is set 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--password dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value="filled"
    />
    <span
      class="dbx-textfield-suffix"
    >
      <button
        class="dbx-button dbx-button--hover-only dbx-button--round dbx-button--s"
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="dbx-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="dbx-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;

exports[`Passwordfield component should render a text input 1`] = `
<div>
  <div
    class="dbx-textfield dbx-textfield--password dbx-textfield--xl dbx-textfield--filled"
  >
    <input
      autocapitalize="none"
      autocomplete="off"
      autocorrect="false"
      spellcheck="false"
      type="password"
      value="test123"
    />
    <span
      class="dbx-textfield-suffix"
    >
      <button
        class="dbx-button dbx-button--hover-only dbx-button--round dbx-button--s"
        title="Passwort im Klartext anzeigen"
        type="button"
      >
        <svg
          class="dbx-icon"
          height="24"
          role="img"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 5c3.688 0 6.406 2.188 9.781 6.375A.955.955 0 0122 12a.955.955 0 01-.219.625C18.406 16.844 15.688 19 12 19c-3.469 0-5.938-1.688-9.781-6.375A.955.955 0 012 12c0-.219.063-.438.219-.625C5.594 7.187 8.313 5 12 5zm0 2c-2.531 0-4.688 1.406-7.719 5C7.22 15.5 9.25 16.906 12 17c2.531 0 4.688-1.406 7.719-5C16.78 8.5 14.75 7.094 12 7zm0 2a3 3 0 11-.002 6.002A3 3 0 0112 9zm0 2c-.563 0-1 .469-1 1 0 .563.438 1.031 1 1.031.563 0 1-.469 1-1.031 0-.531-.438-1-1-1z"
            fill="currentColor"
            fill-rule="evenodd"
          />
        </svg>
        <span
          aria-hidden="false"
        >
          Passwort im Klartext anzeigen
        </span>
      </button>
      <span
        class="dbx-inline-spacer"
      >
         
      </span>
    </span>
  </div>
  <span
    class="dbx-inline-spacer"
  >
     
  </span>
</div>
`;
